File: app/src/main/java/com/example/taxconnect/adapter/TimeSlotAdapter.java
package com.example.taxconnect.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.taxconnect.R;

import java.util.List;

public class TimeSlotAdapter extends RecyclerView.Adapter<TimeSlotAdapter.TimeSlotViewHolder> {

    private final List<String> timeSlots;
    private int selectedPosition = -1;
    private final OnTimeSelectedListener listener;

    public interface OnTimeSelectedListener {
        void onTimeSelected(String timeSlot);
    }

    public TimeSlotAdapter(List<String> timeSlots, OnTimeSelectedListener listener) {
        this.timeSlots = timeSlots;
        this.listener = listener;
    }

    @NonNull
    @Override
    public TimeSlotViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_time_slot, parent, false);
        return new TimeSlotViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull TimeSlotViewHolder holder, int position) {
        holder.bind(timeSlots.get(position), position);
    }

    @Override
    public int getItemCount() {
        return timeSlots.size();
    }

    class TimeSlotViewHolder extends RecyclerView.ViewHolder {
        TextView tvTimeSlot;

        TimeSlotViewHolder(View itemView) {
            super(itemView);
            tvTimeSlot = itemView.findViewById(R.id.tvTimeSlot);
        }

        void bind(String timeSlot, int position) {
            tvTimeSlot.setText(timeSlot);

            if (selectedPosition == position) {
                tvTimeSlot.setBackgroundTintList(itemView.getContext().getColorStateList(R.color.primary));
                tvTimeSlot.setTextColor(itemView.getContext().getColor(android.R.color.white));
            } else {
                tvTimeSlot.setBackgroundTintList(itemView.getContext().getColorStateList(R.color.slate_100));
                tvTimeSlot.setTextColor(itemView.getContext().getColor(R.color.slate_700));
            }

            itemView.setOnClickListener(v -> {
                int previous = selectedPosition;
                selectedPosition = position;
                notifyItemChanged(previous);
                notifyItemChanged(selectedPosition);
                listener.onTimeSelected(timeSlot);
            });
        }
    }
}

File: tax/app/src/main/java/com/example/taxconnect/features/ca/TimeSlotAdapter.kt
package com.example.taxconnect.features.ca

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.example.taxconnect.R

class TimeSlotAdapter(private val onSlotSelected: (String) -> Unit) :
    RecyclerView.Adapter<TimeSlotAdapter.TimeSlotViewHolder>() {

    private var timeSlots: List<String> = ArrayList()
    private var selectedPosition: Int = -1

    fun setTimeSlots(slots: List<String>) {
        this.timeSlots = slots
        this.selectedPosition = -1
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TimeSlotViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_time_slot, parent, false)
        return TimeSlotViewHolder(view)
    }

    override fun onBindViewHolder(holder: TimeSlotViewHolder, position: Int) {
        val slot = timeSlots[position]
        holder.bind(slot, position == selectedPosition)
        
        holder.itemView.setOnClickListener {
            val oldPos = selectedPosition
            selectedPosition = holder.bindingAdapterPosition
            if (selectedPosition != RecyclerView.NO_POSITION) {
                notifyItemChanged(oldPos)
                notifyItemChanged(selectedPosition)
                onSlotSelected(slot)
            }
        }
    }

    override fun getItemCount(): Int = timeSlots.size

    class TimeSlotViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvTimeSlot: TextView = itemView.findViewById(R.id.tvTimeSlot)

        fun bind(slot: String, isSelected: Boolean) {
            tvTimeSlot.text = slot
            
            if (isSelected) {
                tvTimeSlot.backgroundTintList = ContextCompat.getColorStateList(itemView.context, R.color.primary)
                tvTimeSlot.setTextColor(ContextCompat.getColor(itemView.context, R.color.white))
            } else {
                tvTimeSlot.backgroundTintList = ContextCompat.getColorStateList(itemView.context, R.color.slate_100)
                tvTimeSlot.setTextColor(ContextCompat.getColor(itemView.context, R.color.slate_700))
            }
        }
    }
}
